lib/filters/index.js

```diff
@@ -64,8 +64,12 @@ module.exports = ruleSource => {
       if (req.method.toLowerCase() !== method && method !== 'any') {
         return false;
       }
+
+      // Discard any fragments before further processing
+      const mainURI = req.url.split('#')[0];
```

Comment:
Could use the `url` module:
```
> const url = require('url')
> url.parse('/hello/world?some=test').path
'/hello/world?some=test'
> url.parse('/hello/world#hello?some=test').path
'/hello/world'
```

Reply:
Yep, I think it's ideal to use that library (e.g. in case there are other edge cases), but I wanted to keep this PR as light-touch as possible to close the known gap and come back to that after.